src/model/schema/ThunderstoreSchema.ts

```diff
@@ -0,0 +1,131 @@
+import Ajv from "ajv";
+import addFormats from "ajv-formats";
+import schema from "../../../schema.json";
+import schemaSchema from "../../../schemaSchema.json";
+import { GameSelectionDisplayMode } from "../game/GameSelectionDisplayMode";
+import R2Error from "../errors/R2Error";
+
+// Disabled for now, will become relevant when we migrate away from JSON modules.
+export interface Schema {
+    schema_version: string,
+    games: [string:SchemaEntry],
+    communities: [string:ThunderstoreEntry],
+    packageInstallers: [],
+    modloaderPackages: [ModloaderPackage],
+}
+
+export interface SchemaEntry {
+    uuid: string,
+    label: string,
+    meta: EntryMeta,
+    distributions: [EntryDist],
+    thunderstore?: ThunderstoreEntry,
+    r2modman?: EntryR2,
+}
+
+export interface ThunderstoreEntry {
+    displayName: string,
+    discordUrl?: string,
+    wikiUrl?: string,
+    categories: any,
+    sections: any,
+}
+
+export interface ModloaderPackage {
+    identifier: string,
+    rootFolder: string,
+    variant: string,
+}
+
+export interface EntryMeta {
+    displayName: string,
+    iconUrl: string,
+}
+
+export interface EntryDist {
+    platform: DistributionPlatform,
+    identifier?: string,
+}
+
+export interface EntryR2 {
+    internalFolderName: string,
+    dataFolderName?: string,
+    settingsIdentifier: string,
+    packageIndex: string,
+    exclusionsUrl: string,
+    steamFolderName: string,
+    exeNames: [string],
+    gameInstanceType: GameInstanceType,
+    gameSelectionDisplayMode: GameSelectionDisplayMode,
+    packageLoader: string,
+    additionalSearchStrings: [string],
+    installRules: [InstallRule],
+}
+
+export interface InstallRule {
+    route: string,
+    defaultFileExtensions: [string],
+    trackingMethod: string,
+    subRoutes: [InstallRule],
+    isDefaultLocation: boolean,
+}
+
+enum GameInstanceType {
+    GAME = "game",
+    SERVER = "server",
+}
+
+enum DistributionPlatform {
+    STEAM = "steam",
+    STEAM_DIRECT = "steam-direct",
+    EGS = "epic-games-store",
+    OCULUS = "oculus",
+    ORIGIN = "origin",
+    XBOX_GAME_PASS = "xbox-game-pass",
+    OTHER = "other",
+}
+
+export class EcosystemSchema {
+    private static _isValidated: boolean = false;
+
+    /**
+     * Get a validated instance of the ecosystem schema.
+     */
+    static get get() {
```

Comment:
Should this (and r2mm_games) perhaps return a "proper" type instead of "the data as type"? IDK what the correct term is, but now the inferred type e.g. guarantees the existence of `games.garfield-kart-furious-racing` and that seems a bit silly.

Any thoughts on making this getter private and letting the rest of the codebase access stuff through other getters defined below?